lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java

```diff
@@ -1167,74 +1167,78 @@ public String processResource(int type, boolean isWebSite, boolean isCaption) {
-				try {
-					pushed = pushAdvisor();
-					ContentResourceEdit res = contentHostingService.editResource(id);
-					res.setContentType("text/vtt");
-					contentHostingService.commitResource(res, NotificationService.NOTI_NONE);
-				} catch (Exception ignore) {
-					return "no-reference";
-				}finally {
-					if(pushed) popAdvisor();
-				}
+			for(Reference reference : refs){
+				returnMesssage = processSingleResource(reference, type, isWebSite, isCaption);
 			}
 		} else {
 			return "cancel";
 		}
 
-		boolean pushed = false;
+		return returnMesssage;
+	}
+
+	//This method is written to enable user to select multiple Resources from the tool
+	private String processSingleResource(Reference reference,int type, boolean isWebSite, boolean isCaption){
+
+		ToolSession toolSession = sessionManager.getCurrentToolSession();
+		String id  = reference.getId();
+		String description = reference.getProperties().getProperty(ResourceProperties.PROP_DESCRIPTION);
+		String name = reference.getProperties().getProperty("DAV:displayname");
+
+		// URLs are complex. There are two issues:
+		// 1) The stupid helper treats a URL as a file upload. Have to make it a URL type.
+		// I suspect we're intended to upload a file from the URL, but I don't think
+		// any part of Sakai actually does that. So we reset Sakai's file type to URL
+		// 2) Lesson builder needs to know the mime type, to know how to set up the
+		// OBJECT or IFRAME. We send that out of band in the "html" field of the
+		// lesson builder item entry. I see no way to do that other than to talk
+		// to the server at the other end and see what MIME type it claims.
+		String mimeType = reference.getProperties().getProperty("DAV:getcontenttype");
+		if (mimeType.equals("text/url")) {
+			mimeType = null; // use default rules if we can't find it
+			String url = null;
+			// part 1, fix up the type fields
+			boolean pushed = false;
+			try {
+				pushed = pushAdvisor();
+				ContentResourceEdit res = contentHostingService.editResource(id);
+				res.setContentType("text/url");
+				res.setResourceType("org.sakaiproject.content.types.urlResource");
+				url = new String(res.getContent());
+				contentHostingService.commitResource(res, NotificationService.NOTI_NONE);
+			} catch (Exception ignore) {
+				return "no-reference";
+			}finally {
+				if(pushed) popAdvisor();
+			}
+			// part 2, find the actual data type.
+			if (url != null)
+				mimeType = getTypeOfUrl(url);
+		} else if (isCaption) {
+			// sakai probably sees it as a normal text file.
+			// some browsers require the mime type to be right
+			boolean pushed = false;
+			try {
+				pushed = pushAdvisor();
+				ContentResourceEdit res = contentHostingService.editResource(id);
+				res.setContentType("text/vtt");
+				contentHostingService.commitResource(res, NotificationService.NOTI_NONE);
+			} catch (Exception ignore) {
+				return "no-reference";
+			}finally {
+				if(pushed) popAdvisor();
+			}
+		}boolean pushed = false;
```

Comment:
have put all the old code in a separate method so that it can be called for multiple addition of files that is why it looks like this.